sdk/servicebus/service-bus/src/models.ts

```diff
@@ -62,6 +62,11 @@ export interface MessageHandlers {
   processMessage(message: ServiceBusReceivedMessage): Promise<void>;
   /**
    * Handler that processes errors that occur during receiving.
+   *
+   * This handler will be called for any error that occurs in the receiver when
+   *   - receiving the message, or
+   *   - executing your `processMessage` callback, or
+   *   - the receiver automatically completes or abandons the message.
```

Comment:
```suggestion
   *   - receiver is completing the message on your behalf after successfully running your `processMessage` callback and `autoCompleteMessages` is enabled
   *   - receiver is abandoning the message on your behalf if running your `processMessage` callback fails and `autoCompleteMessages` is enabled
   *   - receiver is renewing the lock on your behalf due to auto lock renewal feature being enabled
```

Reply:
Should we add something to reflect that the receiver will automatically retry on all errors?

Reply:
The retry behavior has not been changed in this PR though since the non-retryable classification was removed. We could add some clarification on `subscribe()` ref doc?

Reply:
I agree that the behavior around retry has not changed. When I see docs around the different error scenarios, my next thought is around what should I do about them. Now that we are clarifying the kind of errors, it would also help in clarifying that we do indeed retry on all the errors. And perhaps what the user can do is log them on their side if needed. And maybe point them to the [ref docs on ServiceBusError](https://docs.microsoft.com/en-us/javascript/api/@azure/service-bus/servicebuserror?view=azure-node-latest) to see if there are any errors for which they might want to stop recieving altogether

Reply:
We have some good comments in one sample: https://github.com/Azure/azure-sdk-for-js/blob/83d3df06fd47c31a244b6b3c7e9de9e4c63c4f7e/sdk/servicebus/service-bus/samples-dev/receiveMessagesStreaming.ts

I will adopt them here, and add link to the errors.